pelican/settings.py

```diff
@@ -245,6 +245,22 @@ def get_jinja_environment(settings):
     return settings
 
 
+def _printf_s_to_format_field(printf_string, format_field):
+    """Tries to replace %s with {format_field} in the provided printf_string.
+    Raises ValueError in case of failure.
+    """
+    TEST_STRING = 'PELICAN_PRINTF_S_DEPRECATION'
+    expected = printf_string % TEST_STRING
+
+    result = printf_string.replace('{', '{{').replace('}', '}}') \
+        % '{{{}}}'.format(format_field)
```

Comment:
I think it's cleaner and more understandable to pass `format_field` with `{}`. Then you won't need this and it makes a bit more sense in my opinion:
```python
_printf_s_to_format_field(foo, '{slug}')
# instead of
_printf_s_to_format_field(foo, 'slug')
```

Reply:
Oh, that's true. This is better in that case.